FHEM/90_SIGNALduino_un.pm

```diff
@@ -41,13 +42,17 @@ SIGNALduino_un_Define($$)
   $hash->{lastMSG} =  "";
   $hash->{bitMSG} =  "";
   $hash->{STATE} = "Defined";
-
+  my $name = $hash->{NAME};
+  
   my $iodevice = $a[3] if($a[3]);
   $modules{SIGNALduino_un}{defptr}{$a[2]} = $hash;
   
   my $ioname = $modules{SIGNALduino_un}{defptr}{ioname} if (exists $modules{SIGNALduino_un}{defptr}{ioname} && not $iodevice);
   $iodevice = $ioname if not $iodevice;
   
+  ### Attributes ###
+  $attr{$name}{stateFormat}	= "{ReadingsVal('$name', 'state', '').' | '.ReadingsTimestamp('$name', 'state', '-');}" if( not defined( $attr{$name}{stateformat} ) );
```

Comment:
Da fällt mir ein, dass Stateformat dadurch bei jedem Restart von FHEM auf diesen Wert gesetzt wird und das auch, wenn er modifziert wurde.

Ich bin nicht sicher, wie das am besten abgefangen wird. Als Idee könnten wir darauf prüfen, ob 
`$init_done == 0`  Quelle: https://wiki.fhem.de/wiki/DevelopmentModuleIntro#Wichtige_globale_Variablen_aus_fhem.pl

Reply:
die Lösung für uns ist
> if ( $init_done == 1 ) {
	$attr{$name}{stateFormat}	= "{ReadingsVal('$name', 'state', '').' | '.ReadingsTimestamp('$name', 'state', '-');}" if( not defined( $attr{$name}{stateformat} ) );
  }

somit wird es bei einem Autocreate angelegt und wenn ein User dies löscht und neustartet, so bleibt dies gelöscht. Den Test habe ich nun mehrfach vollzogen.

Reply:
Super :) Macht richtig Spaß so zusammen zu arbeiten :)